gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/vertx/web/handler/impl/internal/mfa/MFAChallengeStep.java

```diff
@@ -48,22 +52,28 @@ public void execute(RoutingContext routingContext, AuthenticationFlowChain flow)
         final Client client = routingContext.get(ConstantKeys.CLIENT_CONTEXT_KEY);
         final MfaFilterContext context = new MfaFilterContext(routingContext, client, factorManager, ruleEngine);
         if (context.isUserSilentAuth()) {
+            LOGGER.debug("User '{}' silently authenticated, skip challenge", context.getEndUserId());
             stopMfaFlow(context, flow);
             return;
         }
         if (!isMfaFlowStopped(context)) {
             if (context.isUserSelectedEnrollFactor()) {
+                LOGGER.debug("User '{}' is enrolling factor, execute Challenge", context.getEndUserId());
                 challenge(context, flow);
-            } else if (!context.isChallengeCompleted() && stepUpRequired(context, client, ruleEngine)) {
-                challenge(context, flow);
-            } else if (isChallengeActive(client)) {
-                switch (getChallengeSettings(client).getType()) {
-                    case REQUIRED -> required(context, flow);
-                    case CONDITIONAL -> conditional(context, flow, client);
-                    case RISK_BASED -> riskBased(context, flow, client);
-                }
             } else {
-                continueFlow(context, flow);
+                boolean stepUpRequired = stepUpRequired(context, client, ruleEngine);
+                if (!context.isChallengeCompleted() && stepUpRequired) {
+                    LOGGER.debug("User '{}' needs to be challenge (stepUp={})", context.getEndUserId(), stepUpRequired);
```

Comment:
User '{}' needs to be challenged
```suggestion
                    LOGGER.debug("User '{}' needs to be challenged (stepUp={})", context.getEndUserId(), stepUpRequired);
```